Goal App/Controllers/HomeViewController.swift

```diff
@@ -45,20 +45,12 @@ class HomeViewController: UIViewController {
 
     private func getGoalsList() {
         showSpinner()
-        firestoreUtil.getGoals { [weak self] response in
-            switch response {
-            case let .success(goalList):
-                self?.originalGoalsList = goalList
-
-            default:
-                self?.originalGoalsList = []
-            }
-            DispatchQueue.main.async { [self] in
-                self?.activeGoals = self?.originalGoalsList?.filter { task in task.completed == false }
-                self?.goalTasks.reloadData()
-                self?.progressCircleSetup()
-                self?.stopSpinner()
-            }
+        Task {
+            originalGoalsList = await firestoreUtil.getGoals()
```

Comment:
Here, I can see that the viewController knows about the `fireStoreUtil` shouldn't the high level layer not know about the low level layer? i.e they should interact with low level layers via abstraction i.e. through protocols I assume that the `fireStoreUtil` is not a protocol is that correct?

Reply:
Yes this is not a protocol